### DISCUSSION TITLE
Construction of a UGRID from Coordinates

### DISCUSSION THREAD

**Discussion by gauravity30** [UXARRAY/uxarray#1015] [Q&A]
Hello,
I am trying to build a GRID from a dataset which contains the coordinates `x`, `y` and `z`. The API mentions the requirement of connectivity variables. Can you clarify what connectivity variables are exactly required ?

Also, is it possible to contruct the connectivity data from coordinates of nodes using any of the inbuilt methods ?

Thanks,
Gaurav

**Comment by rajeeja** [comment]
we had this on pause now, some work along what you mention is here: https://github.com/UXARRAY/uxarray/pull/442

**Reply by gauravity30** [reply]
> we had this on pause now, some work along what you mention is here: https://github.com/UXARRAY/uxarray/pull/442

Hey ! 
Thanks for the reply ! What connectivity data exactly is required by the GRID object constructor along the coordinates.

**Reply by philipc2** [reply]
Hi @airwarriorg91 

We will be prioritizing construction of grids from points in this upcoming iteration. I have looked into using Scipy's [Spherical Voronoi ](https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.SphericalVoronoi.html)as one possibility, in addition to perform triangulations.

**Reply by rajeeja** [reply]
> > we had this on pause now, some work along what you mention is here: #442
> 
> Hey ! Thanks for the reply ! What connectivity data exactly is required by the GRID object constructor along the coordinates.

can you share you input files and use case?

**Reply by gauravity30** [reply]
> > > we had this on pause now, some work along what you mention is here: #442
> > 
> > Hey ! Thanks for the reply ! What connectivity data exactly is required by the GRID object constructor along the coordinates.
> 
> can you share you input files and use case?

I am trying to construct an unstructured grid from the NEK5000 visualization file. I read the elements and the co-ordinates and variables at those coordinates. Now, I have an array of x, y and z, I tried to use the Grid constructor but it required connectivity variables. So, that's why I wanted to know what connectivity variable is required. 

Thanks, 
Gaurav.

**Comment by philipc2** [comment]
Let's continue this discussion in #349
